### PR TITLE
TOOLS/autocrop.lua: use VO crop always

### PR DESCRIPTION
**Pull request by kasper93**
Read this before you submit this pull request:
https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

Reading this link and following the rules will get your pull request reviewed
and merged faster. Nobody wants lazy pull requests.

### CODE REVIEW

**Comment by kasper93** [comment]
@guidocella: Please try with those changes. I ended up simplifying things to avoid corner when multiple crops are provided and so on.

**Comment by christoph-heinrich** [comment]
The documentation for [--reset-on-next-file](https://mpv.io/manual/master/#options-reset-on-next-file) says:
```
Default: Do not reset anything.
```
which isn't accurate anymore with this.

**Comment by kasper93** [comment]
> which isn't accurate anymore with this.

I've updated docs. I was uncertain if this should be tied to this option, but it works nicely. And in fact rotate and crop are so specific to current video playing that it makes more sense to restore them to initial on next one.

**Comment by guidocella** [comment]
It is a bad idea to change the `reset-on-next-file` default because it doesn't reset to 0 but to the CLI or mpv.conf values. For example, `mpv --video-rotate=90 1.mkv 2.mkv 3.mkv 4.mkv`, even if you reset `video-rotate` to 0 on 2.mkv, when switching to 3.mkv and 4.mkv it goes back to 90.

For `autocrop.lua` this can be fixed by setting `file-local-options/video-crop`.

**Comment by kasper93** [comment]
> It is a bad idea to change the reset-on-next-file default because it doesn't reset to 0 but to the CLI or mpv.conf values. For example, mpv --video-rotate=90 1.mkv 2.mkv 3.mkv 4.mkv, even if you reset video-rotate to 0 on 2.mkv, when switching to 3.mkv and 4.mkv it goes back to 90.

If someone sets rotate or crop in CLI or config they want to have it applied for each file. It if very much expected to restore those values on next file. Although I can skip this change if you don't like it and leave it for user to configure.

> For autocrop.lua this can be fixed by setting file-local-options/video-crop.

For autocrop.lua it is indeed better option. But it doesn't change the problem that `reset-on-next-file` resolves when user manually sets crop/rotate during playback.

**Comment by guidocella** [comment]
> If someone sets rotate or crop in CLI or config they want to have it applied for each file. It if very much expected to restore those values on next file. I can skip this change if you don't like it and leave it for user to configure. Although it is quite silly to make users manually unrotate video after manually rotating previous one.

I think the change is useful in most cases but may lead to unexpected behavior if those options are set on the CLI. I guess it's probably worth making. `ab-loop-a` and `ab-loop-b` are also good candidates for the default value.

> For autocrop.lua it is indeed better option. But it doesn't change the problem that `reset-on-next-file` resolves when user manually sets crop/rotate during playback.

It may be more correct for `autocrop.lua` to reset `video-crop` even if users remove it from `reset-on-next-file`, thought if someone intentionally removes it he may want to apply the same autocrop to multiple videos in a row? But this is a rare use case anyway.

**Comment by kasper93** [comment]
I forgot about this PR. @guidocella What is remaining to do here? 

I've added `file-local-options`. Still left the `reset_options` change, because I think it is less confusing. The point is that when new file is loaded with old crop it will be reset to default value always. Else it would be reset only sometimes, for example when it becomes "invalid" it would be lost forever.

Not sure maybe I'm overthinking.

**Comment by guidocella** [comment]
I think it would be useful to normalize `video-crop` `""`, `0` and `0x0` to `""` so scripts don't have to check for both `""` and `0x0`. This hit me too in one of my scripts (though it no longer matters since now `set file-local-options/video-crop 0` is reset to the initial `""` value). `""` makes more sense than `0x0` anyway.

**Comment by kasper93** [comment]
> I think it would be useful to normalize video-crop "", 0 and 0x0 to "" so scripts don't have to check for both "" and 0x0. This hit me too in one of my scripts (though it no longer matters since now set file-local-options/video-crop 0 is reset to the initial "" value). "" makes more sense than 0x0 anyway.

Ok, I made validation more strict. To disallow all `0` and `0x0`, while they could work before, maybe it is indeed better to remove those ambiguous values. Note that you can still do `0x0+x+y` or `0x500`. For better or for worse it will try to infer 0 to some values.

Please try now what do you think. I've also removed `reset-on-next-file` commit. I think it is good, but we can add it later if needed, it is mostly corner case anyway, I doubt it affects a lot of people.

**Comment by guidocella** [comment]
Looks good other than a typo in the second commit subject.

**Review by guidocella — APPROVED** [review]

File: TOOLS/lua/autocrop.lua [hunk 1]
```diff
@@ -323,7 +287,9 @@ function on_toggle()
     end
 
     -- Cropped => Remove it.
-    if remove_filter(labels.crop) then
+    local crop = mp.get_property("file-local-options/video-crop")
+    if crop ~= "" then
```
[the remark this answers is not in the document]
Comment (kasper93) [reply TOOLS/lua/autocrop.lua]: I don't understand, but I only started to see this comment after pushing again. Fixed now.

**Pull request merged**